src/analyzer.rs

```diff
@@ -91,12 +91,19 @@ impl<'src> Analyzer<'src> {
     })
   }
 
-  fn analyze_recipe(&self, recipe: &UnresolvedRecipe<'src>) -> CompilationResult<'src, ()> {
+  fn analyze_recipe(
+                    &self,
+                    recipe: &UnresolvedRecipe<'src>
+                  ) -> CompilationResult<'src, ()> {
+
     if let Some(original) = self.recipes.get(recipe.name.lexeme()) {
-      return Err(recipe.name.token().error(DuplicateRecipe {
-        recipe: original.name(),
-        first:  original.line_number(),
-      }));
+      // TODO if !config.allow_duplicates {
+      //      The problem is that "config" is not defined - I thought it was global
```

Comment:
There do not appear to be any existing CLI Flags that are used during the analysis phase.  I was using the --dry-run process as a guide but it pattern does not fit here.  I do not know the Rust language, so all of my attempts at bringing in the config context to the function have failed.

Reply:
I've been doing some big refactors, so things have moved around a lot, but `Subcommand::run` has the config and creates the `Analyzer`, so it would be the natural place to pass it in:

https://github.com/casey/just/blob/7efb82f4cb6cefa98a543da4e46f458fd6aeb7ed/src/subcommand.rs#L61

Rust is pretty brutal, so this might be a hard first PR!